compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -482,9 +482,32 @@ impl<'a> AstValidator<'a> {
         }
     }
 
-    /// Reject C-variadic type unless the function is foreign,
-    /// or free and `unsafe extern "C"` semantically.
+    /// Reject invalid C-variadic types.
+    ///
+    /// C-variadics must be:
+    /// - Non-const
+    /// - Either foreign, or free and `unsafe extern "C"` semantically
     fn check_c_variadic_type(&self, fk: FnKind<'a>) {
+        let Some(variadic_span) = fk
+            .decl()
+            .inputs
+            .iter()
+            .find(|arg| matches!(arg.ty.kind, TyKind::CVarArgs))
```

Comment:
you could also collect all of them into a `vec` and make the variadic label point to all of them at once.

Reply:
Updated the PR to collect into a vec as suggested